web/lib/api/metrics/getRequestCount.ts

```diff
@@ -0,0 +1,31 @@
+import { Result } from "../../result";
+import { dbExecute } from "../db/dbExecute";
+import { buildFilter, FilterNode } from "./filters";
+
+export interface Count {
+  count: number;
+}
+export async function getRequestCount(
+  filter: FilterNode,
+  user_id: string
+): Promise<Result<number, string>> {
+  const query = `
+SELECT 
+  COUNT(*) AS count
+ FROM request
+   LEFT JOIN response ON response.request = request.id
+   LEFT JOIN user_api_keys ON user_api_keys.api_key_hash = request.auth_hash
+WHERE (
+  user_api_keys.user_id = '${user_id}'
+  AND (${buildFilter(filter)})
```

Comment:
We're clear from SQL injection because
1. We force a filter to the current user (is that user_id?) with an AND. Why are we not using the supabase auth method
2. Because of (1), it doesn't matter if `buildFilter` is all
is that correct?

Reply:
1. Correct, and we are using the supabase auth to grab the user ID and then perform RBAC. We dont need to use Supabase auth because pg is authenticated not as the user. It is a good question tho... there might be a way to do that 🤔 I will follow up with the Supabase team Monday and ask them what they think

2. If build filter is `all` then the only filter applied, is the user id